File: C#/Maximum_Subarray.cs
public class Solution {
    public int MaxSubArray(int[] nums) {
        int maxSoFar = nums[0];
        int currentMax = nums[0];
        
        for (int i = 1; i < nums.Length; i++) {
            currentMax = System.Math.Max(nums[i], currentMax + nums[i]);
            maxSoFar = System.Math.Max(maxSoFar, currentMax);
        }
        
        return maxSoFar;
    }
}

File: Python/Maximum_Subarray.py
class Solution:
    def maxSubArray(self, nums: list[int]) -> int:
        max_so_far = float('-inf')
        current_max = 0
        
        for num in nums:
            current_max += num
            if max_so_far < current_max:
                max_so_far = current_max
            if current_max < 0:
                current_max = 0
        return max_so_far